src/main/java/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package service.impl;
+
+import dto.UserRegistrationDto;
+import dto.UserResponseDto;
+import entity.User;
+import repository.UserRepository;
+import repository.impl.UserRepositoryImpl;
+import service.UserService;
+
+import java.util.List;
+import java.util.Optional;
+
+public class UserServiceImpl implements UserService {
+
+    private static final UserRepository USER_REPOSITORY = new UserRepositoryImpl();
+
+    @Override
+    public UserResponseDto registerUser(UserRegistrationDto dto) {
+        User user = USER_REPOSITORY.save(dto);
+        return user.toUserResponseDto();
+    }
+
+    @Override
+    public Optional<UserResponseDto> getUserById(Long id) {
+        return USER_REPOSITORY.findById(id);
+    }
+
+    @Override
+    public List<UserResponseDto> getUsers() {
+        return USER_REPOSITORY.findAll();
+    }
```

Comment:
Please reutrun `List<User>` from repository, then convert it to List<UserResponseDto>.

```suggestion
@Override
public List<UserResponseDto> getUsers() {
    return userRepository.findAll()
                         .stream()
                         .map(UserMapper::toUserResponseDto)
                         .toList();
}
```